vignettes/introduction.Rmd

```diff
@@ -25,6 +26,173 @@ knitr::opts_chunk$set(
+"automatic" selection, which starts with `"L-BFGS-B"` and proceeds through the 
+other choices if optimization fails to converge. Other choices are "BFGS", "CG", and
+"nlminb". 
+
+```{r common-changes-optim}
+fit_opt <- mmrm(
+  formula = FEV1 ~ RACE + ARMCD * AVISIT + us(AVISIT | USUBJID),
+  data = fev_data,
+  optimizer = "BFGS"
+)
+
+print(fit_opt)
+```
+
+## Covariance Structure
+
+Covariance structures supported by the `mmrm` are being continuously developed. 
+For a complete list and description please visit the [vignette in the package website](https://openpharma.github.io/mmrm/main/articles/covariance.html). Below
+we see the function call for homogeneous compound symmetry (`cs`). 
+
+```{r common-changes-cov}
+fit_cs <- mmrm(
+  formula = FEV1 ~ RACE + ARMCD * AVISIT + cs(AVISIT | USUBJID),
+  data = fev_data,
+  reml = FALSE
+)
+
+print(fit_cs)
+```
+
+# Extraction of model features
+
+Similar to model objects created in other packages, components of `mmrm` and
+`mmrm_tmb` objects can be accessed with standard methods.  Additionally, 
+`component()` is provided to allow deeper and more precise access for those 
+interested in digging through model output.   
+Complete documentation of standard model output methods supported for `mmrm_tmb`
+objects [can be found at the package website.](https://openpharma.github.io/mmrm/main/reference/mmrm_tmb_methods.html)
+
+## Summary extraction
+
+The `summary` method for `mmrm` objects provides easy access to frequently needed
+model components. 
+
+```{r extraction-summary-fit}
+fit <- mmrm(
+  formula = FEV1 ~ RACE + ARMCD * AVISIT + us(AVISIT | USUBJID),
+  data = fev_data
+)
+
+fit_summary <- summary(fit)
+```
+
+From this summary object, you can easily retrieve the coefficients table.
+
+```{r extraction-summary-fit-coef}
+fit_summary$coefficients
+```
+
+Other model parameters and metadata available in the summary object is as follows:
+
+```{r extraction-summary-fit-str}
+str(fit_summary)
+```
+
+## Other components  
+
+Specific model quantities not supported by methods [can be retrieved with the 
+`component()` function.](https://openpharma.github.io/mmrm/main/reference/component.html)
+The default will output all supported components. 
+
+```{r extraction-summary-component}
+str(component(fit))
+```
+
+# Lower level functions
+
+## Low-level mmrm
+
+The lower level function which is called by `mmrm()` is `h_mmrm_tmb()`. This 
+function is exported and can be used directly. It is similar to `mmrm()` but lacks
+some post-processing and support for special d.f. calculations. However, it exposes
+an argument for fine control over optimization which may be needed by some
+users.
+
+```
+h_mmrm_tmb <- function(formula,
+  data,
+  reml = TRUE,
+  control = h_mmrm_tmb_control()
+)
+
+h_mmrm_tmb_control <- function(optimizer = stats::nlminb,
+  optimizer_args = list(),
+  optimizer_control = list(),
+  start = NULL,
+  accept_singular = TRUE
+)
```

Comment:
I would give again instead here example function calls, not cite the function signature

Reply:
done